### ISSUE TITLE
Make WM_MO_* flags per-song instead of global

### ISSUE THREAD

**Issue by sezero** [Mindwerks/wildmidi#130]
We should make most (if not all) WM_MO_ flags per-song instead of global.
Same can be said for sample rate and volume.

Don't know if this can be 0.4.0 material or for a later release, so someone else should set the milestone for it.

**Comment by chrisisonwildcode** [comment]
I'll have to check but I am sure all the WM_MO flags can be set per midi ... 

Volume can be made per midi by adding another API ... Master volume needs to stay. Alternatively an app processing the library output could adjust the returned audio data.

Sample rate cannot be per midi at this point in time. There is no reasonable excuse at this time as to why you would need per midi sample rates. Also implementing such a feature would mean some potential uses would require much more memory to be used for resampling purposes in some cases.

Remember to keep in mind that some functionality is better off in the software using the library.

**Comment by chrisisonwildcode** [comment]
Ok, I'll be modifying some fuctionality of WildMidi_SetOptions() ... Here is what I suggest and ask for your feedback please ...

If WildMidi_SetOptions is called and resampling has not occurred yet then all WM_MO_* options can be set. However, if resampling has occurred then the current behaviour stands. This means WM_MO_SAVEASTYPE0 /WM_MO_ROUNDTEMPO/WM_MO_STRIPSILENCE can be set before a call to WildMidi_GetOutput, but can not be set once resampling has started to occur.

Doing this also means we could remove the options part from WildMidi_Init() ...
Thoughts?

**Comment by chrisisonwildcode** [comment]
still waiting on thoughts on this

**Comment by psi29a** [comment]
@sezero you ok with making this 0.4.1 instead of 0.4? I really want 0.4 out the door. It has been a few years now. ;)

**Comment by sezero** [comment]
> @sezero you ok with making this 0.4.1 instead of 0.4?
> I really want 0.4 out the door. It has been a few years now. ;)

Of course.

**Comment by sezero** [comment]
I won't bother myself with with this anymore - `WONTFIX`
